jwst/photom/photom.py

```diff
@@ -497,6 +503,17 @@ def calc_miri(self, ftab):
             # Update the science dq
             self.input.dq = np.bitwise_or(self.input.dq, ftab.dq)
 
+            # Check if reference file contains time dependent correction
+            if np.any(ftab.timecoeff_ch1['binwave']) and self.mrs_time_correction:
```

Comment:
The check as written assumes `ftab.timecoeff_ch1` exists. If that's not always the case it should include also a check like `if hasattr(ftab, 'timecoeff_ch1`)`

Reply:
Not sure the `hasattr()` approach is correct. Because it's defined in the datamodel schema, `hasattr` is likely to always return true, even if the array/extension doesn't exist in the input file that was loaded. Suggest using the `getarray_noinit` method in the model class, within a try/except block. If the array isn't in the file, it'll fail.

Reply:
the hasattr does not work - it is always true because it is in the data models. I tried that first